nextcord/ui/item.py

```diff
@@ -2,9 +2,11 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, Callable, Coroutine, Generic, Optional, Tuple, TypeVar
+from typing import TYPE_CHECKING, Any, Callable, Coroutine, Generic, Optional, Tuple, TypeVar, Union
 
-from ..interactions import ClientT, Interaction
+from ..interactions.base import ClientT, Interaction
+from ..interactions.message_component import MessageComponentInteraction
+from ..interactions.modal_submit import ModalSubmitInteraction
```

Comment:
`ItemCallbackType` should probably use the component interaction class not the base one:
```python
Argument of type "(self: Self@View, button: Button[Unknown], interaction: CustomComponentInteraction) -> Coroutine[Any, Any, None]" cannot be assigned to parameter of type "ItemCallbackType[Button[V@button], ClientT@button]"
  Type "(self: Self@View, button: Button[Unknown], interaction: CustomComponentInteraction) -> Coroutine[Any, Any, None]" cannot be assigned to type "ItemCallbackType[Button[V@button], ClientT@button]"
    Parameter 3: type "Interaction[ClientT@button]" cannot be assigned to type "CustomComponentInteraction"
      "Interaction[ClientT@button]" is incompatible with "CustomComponentInteraction"Pylance[reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)
```